File: frontend/lib/components/point.dart
class Point {
  final int id;
  final String userId;
  final String latitude;
  final String longitude;
  final DateTime timestamp;
  final String description;
  final String category;
  final String event;
  int votes;

  Point({
    required this.id,
    required this.userId,
    required this.latitude,
    required this.longitude,
    required this.timestamp,
    required this.description,
    required this.category,
    required this.event,
    required this.votes,
  });

  factory Point.fromJson(Map<String, dynamic> json) {
    return Point(
      id: json['id'] ?? 0, // Default value or handle null appropriately
      userId: json['userId'],
      latitude:
          json['latitude'] ?? '', // Default value or handle null appropriately
      longitude:
          json['longitude'] ?? '', // Default value or handle null appropriately
      timestamp: DateTime.parse(json['timestamp']),
      description: json['description'] ??
          '', // Default value or handle null appropriately
      category:
          json['category'] ?? '', // Default value or handle null appropriately
      event: json['event'],
      votes: json['votes'],
    );
  }
}

File: backend/src/main/java/com/safezones/safezones/Model/Point.java
package com.safezones.safezones.Model;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.*;

@Entity
@Getter
@Setter
public class Point {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String userId;
    private String latitude;
    private String longitude;
    private String description;
    private String category;
    private String event;
    private Long votes;
    private LocalDateTime timestamp;

    @ManyToMany
    @JoinTable(
            name = "user_point_likes",
            joinColumns = @JoinColumn(name = "id"),
            inverseJoinColumns = @JoinColumn(name = "userId")
    )
    @JsonManagedReference
    private Set<User> likedByUsers = new HashSet<>();

    public Set<User> getLikedByUsers() {
        return likedByUsers;
    }

    public void setLikedByUsers(Set<User> likedByUsers) {
        this.likedByUsers = likedByUsers;
    }

}
